src/Action/Bulk/BaseAction.php

```diff
@@ -185,8 +184,8 @@ protected function _stopped(Subject $subject)
     /**
      * Handle a bulk event
      *
-     * @param \Cake\ORM\Query $query The query to act upon
+     * @param \Cake\ORM\Query|null $query The query to act upon
      * @return bool
      */
-    abstract protected function _bulk(Query $query);
+    abstract protected function _bulk(Query $query = null);
```

Comment:
Perhaps the extending classes should be fixed instead?  Calling this method without a query instance would make no sense.